tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -1739,35 +1759,50 @@ static void test_device_pb_remote_server_ncrp_second_time(void)
-		.test_pre_init_f = test_back_channel_pre_init,                 \
-		.test_tick_f = bt_mesh_test_timeout,                           \
-		.test_main_f = test_##role##_##name,                           \
-		.test_delete_f = test_terminate                                \
-	}
+/* Test cases by default will run over PB_ADV*/
+#define TEST_CASE(role, name, description)                                                         \
+	{.test_id = "prov_" #role "_" #name,                                                       \
+	 .test_descr = description,                                                                \
+	 .test_post_init_f = test_##role##_init,                                                   \
+	 .test_tick_f = bt_mesh_test_timeout,                                                      \
+	 .test_main_f = test_##role##_##name,                                                      \
+	 .test_delete_f = test_terminate}
+/* Test cases that will run over either PB_ADV or PB_GATT */
+#define TEST_CASE_BEARER(role, bearer, name, description)                                          \
+	{.test_id = "prov_" #role "_" #bearer "_" #name,                                           \
+	 .test_descr = description,                                                                \
+	 .test_post_init_f = test_##role##_init,                                                   \
+	 .test_tick_f = bt_mesh_test_timeout,                                                      \
+	 .test_main_f = test_##role##_##name,                                                      \
+	 .test_delete_f = test_terminate}
+/* Test cases that will run over either PB_ADV or PB_GATT */
+#define TEST_CASE_WBACKCHANNEL(role, bearer, name, description)                                    \
+	{.test_id = "prov_" #role "_" #bearer "_" #name,                                           \
+	 .test_descr = description,                                                                \
+	 .test_post_init_f = test_##role##_init,                                                   \
+	 .test_pre_init_f = test_back_channel_pre_init,                                            \
+	 .test_tick_f = bt_mesh_test_timeout,                                                      \
+	 .test_main_f = test_##role##_##name,                                                      \
+	 .test_delete_f = test_terminate}
 
 static const struct bst_test_instance test_connect[] = {
-	TEST_CASE(device, pb_adv_no_oob,
-		  "Device: pb-adv provisioning use no-oob method"),
-	TEST_CASE_WBACKCHANNEL(device, pb_adv_oob_auth,
-		  "Device: pb-adv provisioning use oob authentication"),
-	TEST_CASE_WBACKCHANNEL(device, pb_adv_oob_public_key,
-		  "Device: pb-adv provisioning use oob public key"),
-	TEST_CASE(device, pb_adv_reprovision,
-		  "Device: pb-adv provisioning, reprovision"),
 	TEST_CASE(device, unresponsive,
 		  "Device: pb-adv provisioning, stops and resumes responding to provisioning"),
+#if defined(CONFIG_BT_MESH_PB_ADV)
```

Comment:
A nitpick maybe. What if we remove ifdefs here and place same test for each bearer next to each other? PB-GATT varians should still compile, but it will add more visibility of what is covered for each bearer, e.g.:
```
TEST_CASE_BEARER(device, pb_adv, no_oob, "Device: pb-adv provisioning use no-oob method"),
TEST_CASE_BEARER(device, pb_gatt, no_oob, "Device: pb-gatt provisioning use no-oob method"),
```
I could also suggest to use a single macro that creates 2 entries for both bearers. But this maybe redundant as we still need a macro for a specific bearer.

Reply:
I made device names the same for both bearer case. Yet, the test case ids we in test scripts will have the bearer name in it so we know which bearer failed.